client/landing/stepper/declarative-flow/internals/steps-repository/domain-search/index.tsx

```diff
@@ -69,16 +69,23 @@ const DomainSearchStep: StepType< {
 	submits: UseMyDomain | StepSubmission;
 } > = function DomainSearchStep( { navigation, flow } ) {
 	const userSiteCount = useSelector( getCurrentUserSiteCount );
-	const site = useSite();
+	const sourceSlug = useQuery().get( 'sourceSlug' );
+	const site = useSite( sourceSlug ?? undefined );
 	const siteSlug = useSiteSlugParam();
 	const siteId = useSiteIdParam();
 	const initialQuery = useQuery().get( 'new' ) ?? '';
 	const tldQuery = useQuery().get( 'tld' );
 	const source = useQuery().get( 'source' );
 	const backTo = useQuery().get( 'back_to' );
 
+	const siteFragment = siteSlug || sourceSlug;
+
 	// eslint-disable-next-line no-nested-ternary
-	const currentSiteUrl = site?.URL ? site.URL : siteSlug ? `https://${ siteSlug }` : undefined;
+	const currentSiteUrl = site?.URL
```

Comment:
We can't use the `sourceSlug` as the current site URL here because it means the user will keep the existing site as the copied site URL and that's not correct:

<img width="1103" height="154" alt="image" src="https://github.com/user-attachments/assets/f0082d4f-b17b-4ac8-824d-8ce544e30987" />